### ISSUE TITLE
Incorrect exponential regime dispatch for Cl(0,q) (Hyperbolic -> Elliptic)

### ISSUE THREAD

**Issue by Concode0** [Concode0/Versor#12]
### TL;DR

CliffordAlgebra._exp_regime statically misclassifies Cl(0, q, 0) as hyperbolic, so _exp_bivector_closed uses the cosh/sinhc branch on a bivector whose square is actually negative. The resulting "rotor" is not in the Spin group: R * ~R ≠ 1 and sandwich products fail to preserve norm.

### Affected Area

core/algebra.py - CliffordAlgebra.__init__ regime classifier

### Algebra Signature

Cl(0, q, 0) for q >= 2

### What's Happening

```shell
This infers the bivector-square sign from the signature layout, but basis-bivector squares follow (e_i e_j)^2 = -s_i . s_j with s_k are in {+1, -1, 0}
```

### Expected Behavior

exp(-B/2) must return a unit rotor for every pure bivector B in every signature: R * ~R == 1 and |R x ~R| == |x|. In Cl(0, q, 0) the correct closed form is the elliptic branch, since alpha = Sum (b_k)^2 . (-1) <= 0.

### Minimal Reproduction

```python
import torch
from core.algebra import CliffordAlgebra

alg = CliffordAlgebra(0, 3, 0, device='cpu')
torch.manual_seed(0)
B = torch.randn(4, alg.dim) * alg.grade_masks[2].to(torch.float32)
R = alg.exp(-0.5 * B)
RRt = alg.geometric_product(R, alg.reverse(R))
ident = torch.zeros_like(RRt); ident[..., 0] = 1.0
print('max |R*~R - 1| =', (RRt - ident).abs().max().item())
# ~2.8e+01 — should be ~1e-7
```

### Checks Already Tried (for instability / performance)

- [ ] Reduced learning rate
- [ ] Reduced bivector norm clip (`optimizer.bv_clip`)
- [ ] Switched to standard Adam
- [ ] Reduced model depth / channels
- [ ] Used `.reshape()` instead of `.view()` after multi-rotor ops
- [x] Passed `device='cpu'` explicitly to `CliffordAlgebra`
- [x] Tried a different `exp_policy` (`auto` / `fast` / `exact`)

### Device

- [ ] CPU
- [ ] CUDA
- [ ] Apple Silicon (MPS)

### Versor Commit / Python / PyTorch

78c57d3, Python 3.11.12, PyTorch 2.10.0
